lib/matplotlib/dates.py

```diff
@@ -605,16 +604,7 @@ def drange(dstart, dend, delta):
 
 
 def _wrap_in_tex(text):
-    p = r'([a-zA-Z]+)'
-    ret_text = re.sub(p, r'}$\1$\\mathdefault{', text)
-
-    # Braces ensure symbols are not spaced like binary operators.
-    ret_text = ret_text.replace('-', '{-}').replace(':', '{:}')
-    # To not concatenate space between numbers.
-    ret_text = ret_text.replace(' ', r'\;')
-    ret_text = '$\\mathdefault{' + ret_text + '}$'
-    ret_text = ret_text.replace('$\\mathdefault{}$', '')
-    return ret_text
+    return r"{\fontfamily{\familydefault}\selectfont " + text + "}"
```

Comment:
My matplotlib/latex integration ignorance is vast, but why do we need to do all the font selection here?  (I guess I'm really asking why this needs to wrapped at all)

Reply:
Because tick formatters can only set the ticklabel *text*, but not the ticklabel font (and moreover the way latex integration is set up right now, we don't even respect the font setting on Text object, but only the global `rcParams["font.family"]`, which is something that we should fix too...

So forcing the font via tex commands is using a big hammer to just go around all these limitations.

Reply:
But I mean, why isn't the default latex font acceptable?  If I just put "23-Dec-2022" in a Latex document, that looks fine?

Reply:
See https://github.com/matplotlib/matplotlib/issues/2294#issuecomment-424121056: this is because we default to forcing non-math to sans-serif, which looks inconsistent with the serif numeric ticks.  I agree with your point at https://github.com/matplotlib/matplotlib/issues/22350#issuecomment-1024950236 that defaulting to sans-serif tex may not have been the best choice, but changing that would be somewhat trickier.